Sources/Conformance/conformance.pb.swift

```diff
@@ -245,18 +245,24 @@ struct Conformance_ConformanceRequest: SwiftProtobuf.Message, SwiftProtobuf.Prot
 
   init() {}
 
-  public mutating func _protoc_generated_decodeField<T: SwiftProtobuf.FieldDecoder>(setter: inout T, protoFieldNumber: Int) throws {
-    switch protoFieldNumber {
-    case 1, 2: try payload.decodeField(setter: &setter, protoFieldNumber: protoFieldNumber)
-    case 3: try setter.decodeSingularField(fieldType: Conformance_WireFormat.self, value: &requestedOutputFormat)
+  public mutating func _protoc_generated_decodeMessage<D: SwiftProtobuf.Decoder>(decoder: inout D) throws {
```

Comment:
I do want to revisit this post-1.0.

The current design puts everything on the struct or everything in the CoW storage class; it should be feasible to use both strategies in the same message for different fields.

Associated with this is a performance exploration -- intuitively, it seems that very large structs would be a performance problem so that we should limit the total number of fields directly on the struct and push others into the heap storage.  Protobuf has a general notion that field numbers <16 should be faster than others, so this may be one of the considerations.

Post-1.0, I want to look at overhauling the generator code to introduce a modeling layer that represents the logical structure.  Then the generator logic looks like:  build the model from the descriptor, then walk the model to emit the code.  This is essentially the same idea as the C++ distinction between 'DescriptorProto' and 'Descriptor' -- the latter is what I'm calling the "model".  If done carefully, this model would also simplify separately building the JSON/Text support and simplify tools such as gRPC.

Among other things, I would hope that introducing this modeling layer would make it easier to experiment with field storage organization, by providing a clean place to handle per-field storage decisions.